tofd/pars/R3BTofDCal2HitPar.cxx

```diff
@@ -297,1099 +352,2668 @@ void R3BTofDCal2HitPar::Exec(Option_t* option)
-                                                               para->GetPar4Walk(),
-                                                               para->GetPar5Walk());
-                                        top_ns = top_ns - walk(top_tot,
-                                                               para->GetPar1Walk(),
-                                                               para->GetPar2Walk(),
-                                                               para->GetPar3Walk(),
-                                                               para->GetPar4Walk(),
-                                                               para->GetPar5Walk());
-                                        }*/
-
-                    auto posToT =
-                        para->GetLambda() * log((top_tot + para->GetToTOffset2()) / (bot_tot + para->GetToTOffset1()));
-
-                    // fill control histograms
-                    // fhTot1vsTot2[iPlane - 1][iBar - 1]->Fill(top_tot, bot_tot);
-                    fhSqrtQvsPosToT[iPlane - 1][iBar - 1]->Fill(posToT, sqrt(top_tot * bot_tot));
-
-                    // Time differences of one paddle
-                    fhTdiff[iPlane - 1]->Fill(iBar, tdiff);
-
-                    // Time of hit
-                    auto THit = (bot_ns + top_ns) / 2. - para->GetSync();
-
-                    // Sync of one plane
-                    fhTsync[iPlane - 1]->Fill(iBar, THit);
-                }
-
-                // prepare double exponential fit
-                if (!fTofdSmiley && fTofdQ > 0.1 && fParameter > 1)
-                {
-                    LOG(debug) << "Prepare histo for double exponential fit";
-                    auto par = fHitPar->GetModuleParAt(iPlane, iBar);
-                    if (!par)
-                    {
-                        R3BLOG(error,
-                               "Hit par not found, Plane: " << top->GetDetectorId() << ", Bar: " << top->GetBarId());
-                        continue;
-                    }
-
-                    // calculate y position
-                    auto pos = ((bot_ns + par->GetOffset1()) - (top_ns + par->GetOffset2())) * par->GetVeff();
-
-                    // fill fitting histograms and smiley histogram
-                    fhTot1vsPos[iPlane - 1][iBar - 1]->Fill(pos, bot_tot);
-                    fhTot2vsPos[iPlane - 1][iBar - 1]->Fill(pos, top_tot);
-                }
-
-                // std::cout<< " hola 10" << fTofdZ << std::endl;
-
-                // prepare charge fit / quench correction
-                if (fTofdZ == true && fParameter > 1)
-                {
-                    // std::cout<< " hola 20" << std::endl;
-                    LOG(debug) << "Prepare histo for quenching correction";
-                    // get parameter
-                    auto par = fHitPar->GetModuleParAt(iPlane, iBar);
-                    if (!par)
-                    {
```

Comment:
Redeclaring auto par = fHitPar->GetModuleAt(iPlane, iBar) again. Could be added into 1 (fTofdQ > 0 && fParameter > 1)